### PR TITLE
GEOMESA-1682 Improve functionality of Leaflet templating in geomesa-jupyter

### PR DESCRIPTION
**Pull request by cfkelly**
Signed-off-by: Charles Kelly <charles.kelly@ccri.com>

### CODE REVIEW

**Comment by mdzimmerman** [comment]
Unless this is a WIP you're probably gonna want to create a ticket on the GeoMesa JIRA and list it here.

File: geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala [hunk 1]
```diff
@@ -71,64 +146,50 @@ object L {
...
     override def render: String = {
       val coords = ls.getCoordinates.map { c => Array(c.y, c.x).mkString("[", ",", "]") }.mkString("[", ",", "]")
-      s"L.polyline($coords).addTo(map);"
+      s"L.polyline($coords, ${style.render} ).addTo(map);"
     }
   }
 
   // TODO: parameterize base url for js and css
-  def buildMap(layers: Seq[GeoRenderable], center: (Double, Double) = (0,0), zoom: Int = 8) =
+  def buildMap(layers: Seq[GeoRenderable], center: (Double, Double) = (0,0), zoom: Int = 8, path: String = "js") =
```
Comment (mdzimmerman) [inline geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala]: +1

File: geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala [hunk 2]
```diff
@@ -31,6 +40,52 @@ object L {
        """.stripMargin
   }
 
+  /**
+    * Takes a string containing a valid javascript styling function as its argument
+    * @param javascriptFunction The javascript styling function
+    */
+  case class StyleOptionFunction(javascriptFunction: String) extends StyleOption {
+    def render: String = javascriptFunction
```
Comment (mdzimmerman) [inline geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala]: This is great! Can you add some docs here or on the ``README.md`` on how to use it? Specifically, what will be passed to the function, and what should it return?

File: geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala [hunk 3]
```diff
@@ -31,6 +40,52 @@ object L {
...
+     """.stripMargin
+  }
+
+  private case class PointToLayer(style: StyleOption) {
+    val styleOptions = style.render
+    def render: String =
+      s"""
+        |pointToLayer: function(feature, latlng) {
+        |  return L.circleMarker(latlng, {
+        |                                   radius: 5,
```
Comment (cfkelly) [inline geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala]: Will do.

File: geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala [hunk 4]
```diff
@@ -31,6 +40,52 @@ object L {
...
+       |function onClick(e) {
+       |  e.target.openPopup();
+       |}
+       |
+       |function onFeature(feature, layer) {
+       |  var keys = Object.keys(layer.feature.properties);
+       |  var str = "";
+       |  for (var i in keys) {
+       |    var key = keys[i];
+       |    if (key != "geom") {
```
Comment (cfkelly) [inline geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala]: I fixed this issue removing that bit of logic and by adding ```!b.isInstanceOf[Geometry]``` to the call on line 145:
```
sf.getType.getAttributeDescriptors.zip(sf.getAttributes)
   .filter{case(a,b) => b != null && !b.isInstanceOf[Geometry]}
   .map { case (d, a) => propToJson(d, a) }.mkString(sep =",\n")
```

File: geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala (hunk 2 above)
Comment (cfkelly) [inline geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala]: Added a little bit to the ```README.md```. May need to correct the formatting a little.

**Pull request merged**